File: SolverSeged/SolverSeged/TanH.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SolverSeged
{
    class TanH : ActivationStrategy
    {
        public override double Activation(double input)
        {
            return Math.Tanh(input);
        }

        public override double DeActivation(double output)
        {
            return 1 - (output * output);
        }

        public override string ToString()
        {
            return "TanH";
        }
    }

}

File: ProblemClassifier/src/Classes/TanH.java
package Classes;

/**
 * @author Danisovszky Márk
 */
public class TanH extends ActivationFunction {

    @Override
    public double Activation(double input) {
        return Math.tanh(input);
    }

    @Override
    public double DActivation(double input) {
        return 1 - (input * input);
    }
}
